packages/checkout-extension/src/ExtensionProvider.tsx

```diff
@@ -0,0 +1,54 @@
+import { CheckoutService } from '@bigcommerce/checkout-sdk';
+import React, { ReactNode, useReducer } from 'react';
+
+import { ExtensionContext } from './ExtensionContext';
+import { ExtensionService } from './ExtensionService';
+import { isCheckoutExtensionEnabled } from './isCheckoutExtensionEnabled';
+
+export interface ExtensionState {
+    isShowingLoadingIndicator: boolean;
+}
+
+export interface ExtensionAction {
+    type: ExtensionActionType;
+    payload: boolean;
+}
+
+export enum ExtensionActionType {
+    SHOW_LOADING_INDICATOR,
+}
+
+export interface ExtensionProviderProps {
+    checkoutService: CheckoutService;
+    children: ReactNode;
+}
+
+export const ExtensionProvider = ({ checkoutService, children }: ExtensionProviderProps) => {
+    const isExtensionEnabled = () =>
+        isCheckoutExtensionEnabled(checkoutService.getState().data.getConfig()?.checkoutSettings);
+
+    const extensionReducer = (state: ExtensionState, action: ExtensionAction): ExtensionState => {
```

Comment:
In the next clean up PR maybe a good idea to move this out of provider as well :D

Reply:
Sure, I can include it when bumping https://github.com/bigcommerce/checkout-sdk-js/pull/2085.